packages/components/src/range-control/utils.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * External dependencies
+ */
+import { clamp, noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useRef, useEffect, useState } from '@wordpress/element';
+
+/**
+ * A float supported clamp function for a specific value.
+ *
+ * @param {number} value The value to clamp
+ * @param {number} min The minimum value
+ * @param {number} max The maxinum value
+ * @return {number} A (float) number
+ */
+function floatClamp( value, min, max ) {
+	return parseFloat( clamp( value, min, max ) );
+}
+
+/**
+ * Hook to store a clamped value, derived from props.
+ */
+export function useControlledRangeValue( { min, max, value: valueProp = 0 } ) {
+	const [ value, _setValue ] = useState( floatClamp( valueProp, min, max ) );
+	const valueRef = useRef( value );
+
+	const setValue = ( nextValue ) => {
+		_setValue( floatClamp( nextValue, min, max ) );
+	};
```

Comment:
Personally, I don't usually care for `_` prefixing, typically because there's usually a better way to distinguish whatever it is we're otherwise assuming to be "the same". In this case, one is responsible for directly setting the state value. The other sets the state value, but within particular constraints. Thus, I could imagine some naming like `setValue` and something like `setClampedValue`.